[docs] Add draft AI tutorial (#46779)
expo/expo: docs/pages/tutorial/build-with-ai/create-your-first-app.mdx

---
title: Create your first app
description: In this chapter, direct your AI agent to create a new Expo app and see it running on your phone.
---

import { Collapsible } from '~/ui/components/Collapsible';
import { ContentSpotlight } from '~/ui/components/ContentSpotlight';
import { ProgressTracker } from '~/ui/components/ProgressTracker';
import { Terminal } from '~/ui/components/Snippet';
import { Step } from '~/ui/components/Step';

> In this chapter, your agent creates a new Expo app, you get it running on your phone, and you make your first change — the loop you'll repeat for the rest of the tutorial.

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
In this chapter, your agent creates a new Expo app, you get it running on your phone, and you make your first change. This is the loop you'll repeat for the rest of the tutorial.
```

<Step label="1">

## Start your agent in a new folder

Your agent works inside one folder at a time, so create a dedicated folder for this project and start the agent there. In your terminal, run:

> <Terminal cmd={['$ mkdir StickerSmash', '$ cd StickerSmash']} />

Reviewer: ```suggestion
<Terminal
  cmd={['$ mkdir StickerSmash', '', '$ cd StickerSmash']}
  cmdCopy={['$ mkdir StickerSmash && cd StickerSmash']}
/>
```

I think, it'd be easier to copy both commands with a simple copy button click.

Then start your agent in that folder: run `claude` or `codex`. If you're using Cursor, create a new folder named **StickerSmash**, open it with **File** > **Open Folder**, and open the agent panel.

</Step>

<Step label="2">

## Create the app

Paste the following prompt into your agent:

```text Prompt|wrap
Create a new Expo app in this folder (the current directory) by running npx create-expo-app@latest and choosing the SDK 54 template — I will test the app with Expo Go on my phone, which uses SDK 54. After it is created, run the project's reset-project script so we start from a minimal app, and delete the app-example folder that the script leaves behind. Don't start the development server; I will run that myself.
```

> The agent will take a few minutes to download the template and install its dependencies. When it finishes, it should report that the project is ready, with a minimal **app** folder containing a couple of files.

Reviewer: **[suggestion]** Use present tense (`style-guide.md#voice-and-tone`).

```suggestion
The agent takes a few minutes to download the template and install its dependencies. When it finishes, it should report that the project is ready, with a minimal **app** folder containing a couple of files.
```

</Step>

<Step label="3">

## Run the app on your phone

> This is the one command you'll run yourself throughout the tutorial — it starts the development server that streams your app to your phone. Open a **new terminal window** (leave your agent running in the first one), then run:

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
This is the one command you'll run yourself throughout the tutorial. It starts the development server that streams your app to your phone. Open a **new terminal window** (leave your agent running in the first one), then run:
```

> <Terminal cmd={['$ cd StickerSmash', '$ npx expo start']} />

Reviewer: ```suggestion
<Terminal
  cmd={['$ cd StickerSmash', '', '$ npx expo start']}
  cmdCopy={['$ cd StickerSmash && npx expo start']}
/>
```

Same as previous.

A QR code appears in the terminal. To open the app:

- **Android**: open Expo Go and tap **Scan QR code**.
- **iOS**: open the default camera app and point it at the QR code.

> You should see a mostly empty screen on your phone — that's your app. Keep this terminal window open and your phone nearby for the rest of the tutorial.

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
You should see a mostly empty screen on your phone. That's your app. Keep this terminal window open and your phone nearby for the rest of the tutorial.
```

</Collapsible>

</Step>

<Step label="4">

## Make your first change

Now close the loop: ask the agent for a visible change and watch it land on your phone. Paste the following prompt into your agent:

```text Prompt|wrap
Change the home screen so it shows the text "Home screen" in white, centered on a dark background with the color #25292e.
```

**What you should see**: within seconds of the agent finishing, the app on your phone reloads to a dark screen with white "Home screen" text in the middle:

<ContentSpotlight
  alt="App with updated text and background color"
  src="/static/images/tutorial/02-index-screen-changes.png"
  className="max-w-[720px]"
/>

That's the whole workflow — you prompt, the agent builds, your phone shows the result.

</Step>

## When something looks wrong

Sooner or later a result won't match what you asked for. That's normal, and fixing it is part of the workflow:

- **Describe, don't diagnose.** Tell the agent exactly what you see and what you expected: "The text is centered but the background is still white." You don't need to guess why.

> - **Paste errors verbatim.** If your phone shows a red error screen or the terminal prints an error, copy the whole message into the agent. Error messages are written for programmers — your agent is one.

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
- **Paste errors verbatim.** If your phone shows a red error screen or the terminal prints an error, copy the whole message into the agent. Error messages are written for programmers, and your agent is one.
```
